source/runtime_api.cpp

```diff
@@ -178,7 +178,7 @@ void reshade::runtime::get_uniform_variable_name([[maybe_unused]] api::effect_un
 		else if (*size != 0)
 		{
 			*size = variable->name.copy(value, *size - 1);
-			value[*size++] = '\0';
+			value[*size] = '\0';
```

Comment:
Please don't change this behavior: When passing a `nullptr` for value the functions return the buffer size (including the null-terminator), so it is confusing if after passing in a buffer the behavior of the size parameter suddenly changes. It should still return the actual buffer size (including the null-terminator).

Reply:
I thought you might say that, but it is true.
No, it is safe. This commit ([No required to increment the pointer](https://github.com/crosire/reshade/pull/316/commits/aab2429aa1ea528283c0f36c8e62ee618609141a)) will not change this odd behavior.

The story begins when I noticed the behavior that calling this function repeated calls truncate the tail of the string by one byte.
In the size report, it returns the required size up to the null byte assignment, but in the assignment, it returns the size of the string excluding the null byte.
So what is the increment operator incrementing?
You should have written `(*size)++` (at [Change API functions returning a string to automatically null-terminate them again](https://github.com/crosire/reshade/blame/8986e2a814dc9d1acc6cf882061493fccb75f32a/source/addon.cpp#L40)). This is API error.
So, this commit will not change this odd behavior.